### PR TITLE
PW-2415 Added Jackson annotations in models for serialization

### PR DESCRIPTION
**Pull request by saquibsayyad**
**Description**
Added Jackson annotations in the models so that they can be serialized the same way as GSON

**Fixed issue**:  
#340

### CODE REVIEW

**Review by andrei-dediu — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -25,6 +25,7 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <jackson.version>2.11.2</jackson.version>
```
Comment (rkewlani) [inline pom.xml:28]: can you please use 2.9.7?

Comment (saquibsayyad) [reply pom.xml]: done

**Review by andrei-dediu — APPROVED** [review]

**Review by Aleffio — DISMISSED** [review]

**Review by rkewlani — DISMISSED** [review]

**Review by Aleffio — APPROVED** [review]

**Review by andrei-dediu — APPROVED** [review]

**Comment by deepu105** [comment]
@saquibsayyad @andrei-dediu seems like annotations are missing for `SepaDirectDebitDetails`, any reason this class is needed as the `DefaultPaymentMethodDetails` seems to have this field. If not required we should mark it as deprecated IMO

**Comment by saquibsayyad** [comment]
Hi @deepu105 ,

`SepaDirectDebitDetails` has field name same as in the json representation, so Jackson annotations won't be required. 
`DefaultPaymentMethodDetails` is part of old implementation, we will be deprecating it in future.

Thanks,
Saquib

**Comment by deepu105** [comment]
But the data sent from the client side still uses the old format so this doesn't work at the moment and I would assume changing how the front end sends data is a breaking change?

**Comment by saquibsayyad** [comment]
Hi @deepu105,

Both the formats are working for now. `sepa.ibanNumber` and `sepa.ownerName` fields are already deprecated internally and will be removed later. It is replaced by `iban` and `ownerName`.
Thanks,
Saquib

**Pull request merged**